### PR TITLE
chore: update github action workflows

### LINKED ISSUE #18: Refactor jobs in CI workflow

### PR DESCRIPTION
**Pull request by rosdb**
Resolve #18

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -10,17 +10,29 @@ on:
...
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Customize user email
+        run: git config --global user.email "$USER_EMAIL"
+        env:
+          USER_EMAIL: ${{ secrets.DETRA_BOT_EMAIL }}
+
+      - name: Customize user name
+        run: git config --global user.name "detra-bot"
```
Comment (giotramu) [inline .github/workflows/main.yaml:27]: Have you tried to merge the two user configuration steps?

Something like that:

```yaml
- name: Customize GIT user
  run: |
    git config --global user.email "$USER_EMAIL"
    git config --global user.name "detra-bot"
  env:
    USER_EMAIL: ${{ secrets.DETRA_BOT_EMAIL }}
```

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -10,17 +10,29 @@ on:
...
     steps:
-      - uses: actions/checkout@v3
-      - uses: pnpm/action-setup@v2
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Customize user email
+        run: git config --global user.email "$USER_EMAIL"
+        env:
+          USER_EMAIL: ${{ secrets.DETRA_BOT_EMAIL }}
```
Comment (giotramu) [inline .github/workflows/main.yaml:24]: Did you configure the env variable `DETRA_BOT_EMAIL` in the organization secrets?

**Review by giotramu — CHANGES_REQUESTED** [review]

**Comment by giotramu** [comment]
How about reusing the same workflow for the following repositories?

@detra-lab/eslint-config
@detra-lab/prettier-config
@detra-lab/stylelint-config

[Github Actions - Reusing workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow)

Maybe we can do that in another PR? Or do you want to do a spike in this one?

File: .github/workflows/main.yaml (hunk 2 above)
Comment (rosdb) [reply .github/workflows/main.yaml:24]: Yes, I did 👍🏻

**Comment by rosdb** [comment]
> How about reusing the same workflow for the following public repositories?
> 
> @detra-lab/eslint-config @detra-lab/prettier-config @detra-lab/stylelint-config
> 
> [Github Actions - Reusing workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow)
> 
> Maybe we can do that in another PR? Or do you want to do a spike in this one?

@giotramu, I'll do a spike in this one.

**Review by giotramu — APPROVED** [review]

**Comment by giotramu** [comment]
> @giotramu, I'll do a spike in this one.

To track the activity, I've created a [new task](https://github.com/orgs/detra-lab/projects/2/views/1?pane=issue&itemId=31873309).

**Pull request merged**